docs/usage.md

```diff
@@ -0,0 +1,213 @@
+
+## Table of Contents
+
+* [Overview](#overview)
+* [Example Usage - Hello World](#example-usage---hello-world)
+* [Example Usage - Records and Datastores](#example-usage---records-and-datastores)
+* [Example Usage - Bulk Validation](#example-usage---bulk-validation)
+
+## Overview
+
+The main "entry point", or class to instantiate for validation, is `QualityCheck`, which in turn creates a `NACCValidator`, which both validate the schema and returns error information from the validator. `NACCValidator` itself is an extension of [Cerberus' `Validator` class](https://docs.python-cerberus.org/api.html#cerberus.Validator). It can also use an optional `Datastore` object which you can implement to access records in your own database. See [Example Usage - Records and Datastores](#example-usage---records-and-datastores) for more information.
+
+The general workflow is to instantiate a `QualityCheck` object with the schema you want to validate against, and then pass the record to validate to the `validate_record` method. This method returns 4 variables:
+
+| Variable | Type | Description |
+| -------- | ---- | ----------- |
+| `passed` | `bool` | Whether or not the record satisfied all validation rules |
+| `sys_failure` | `bool` | Whether or not a system error occured |
+| `errors` | `dict[str, list[str]]` | Dict of errors encountered keyed by the variable that failed. Empty if no errors encountered for the record. |
+| `error_tree` | `DocumentErrorTree` | A dict-like object of `ValidationError` instances. See [Cerberus' Errors documentation](https://docs.python-cerberus.org/errors.html) for more information. |
+
+`QualityCheck` itself is fairly straightforward - the actual validation logic is handled by `NACCValidator`. There isn't really any reason to use `NACCValidator` directly, but if you decide to just keep in mind the following (e.g. duplicate what `QualityCheck` will handle for you):
+
+* If the records you're validating on have _missing_ fields (e.g. passing an empty `dict` as a "record" as opposed to a `dict` with all fields set to `None`), some of the more complicated rules like `logic` may not entirely behave as expected. The `cast_record` resolves this by setting any missing fields (based on the schema) to `None`, **so you need to call this method before validating**
+* Similarly, if using `primary_key` and `datastore`, those properties will need to be explicitly set on the `NACCValidator` object
+
+## Example Usage - Hello World
+
+The following is a simple example of using this package. The schema is a single rule for the field `hello`, and the only value it can be assigned to is the string `world`.
+
+```python
+from nacc_form_validator import QualityCheck
+
+pk_name = "example_key"
+
+schema = {
+    "hello": {
+        "type": "string",
+        "required": True,
+        "allowed": [
+            "world"
+        ]
+    }
```

Comment:
Add the field "example_key" to the schema and data record. Primary key field should be defined in the schema as a required field and should be present in data.